### ISSUE TITLE
Soft integration of CensuScope lite into HIVE1/2

### ISSUE THREAD

**Issue by penningtonea** [GW-HIVE/CensuScope#5]
Indexing new version of NT is failing in HIVE. We want an external version of the CensuScope tool that can use the preformatted NT from NCBI.

Hadley has created the tool, now we just need the working Docker image.

This first iteration will use the full 20Gb TaxDB inside the Docker image for the sake of speed.

**Comment by kee007ney** [comment]
Blocked: waiting on a running docker image from Hadley.

**Comment by cwoodside1278** [comment]
We need local deployment instructions. @penningtonea will test on Pegasus with slimNT. @HadleyKing updates?

**Comment by cwoodside1278** [comment]
@kee007ney will test it again to see if it is working.

**Comment by kee007ney** [comment]
Front end tool page exists on Comb. Remaining items are:
- Run Hadley's tool at CLI
- Package as HIVEpack
- Map all parameters to front end page

**Comment by kee007ney** [comment]
Rocky update on HIVE servers apparently wiped all dependencies (e.g. git, docker, etc.).
Rebuilding environment.

**Comment by kee007ney** [comment]
@HadleyKing I get an error of `"/test_data": not found` when trying to run the `docker build -t  censuscope .` command. I think we talked about it and you said that it was something like you accidentally added it to your .gitignore file, but I can't find any documentation so I don't remember what the resolution was. Were you going to add the test data to the repo? Or was it a different solution?

**Comment by cwoodside1278** [comment]
@kee007ney This ticket is related to #1 that was just closed.

**Comment by cwoodside1278** [comment]
Jonathon has already built the environment, but needs to troubleshoot it. Still waiting to hear back from @HadleyKing about the test data

**Comment by kee007ney** [comment]
Pull package from here:
https://github.com/GW-HIVE/CensuScope/pkgs/container/censuscope

Hadley will email when it's updated with latest code/documentation.

**Comment by HadleyKing** [comment]
# 🧬 Running CensuScope with Docker Compose

This guide walks you through pulling and running the `ghcr.io/gw-hive/censuscope:2.1.0` Docker image using Docker Compose.

---

## ✅ Prerequisites

- Docker Desktop installed (macOS/Windows/Linux)
- Docker Compose available via `docker compose` command
- A FASTQ file and BLAST database available locally

---

## 📂 Local Directory Structure (Example)

Your directory should look like this:

```bash
your-project/
├── docker-compose.yml
├── temp_dirs/                # (output files will be saved here)
└── your-data/
    ├── query.fastq
    └── blast_db/
        ├── HumanGutDB.nin
        ├── HumanGutDB.nsq
        ├── HumanGutDB.nhr
        └── HumanGutDB.fasta
```

---

## 🐳 Example `docker-compose.yml`

```yaml
version: "3.8"

services:
  censuscope:
    image: ghcr.io/gw-hive/censuscope:2.1.0
    container_name: censuscope_localtest
    command: >
      python lib/censuscope.py
      -i 5
      -s 10
      -t family
      -q /app/data/query.fastq
      -d /app/data/blast_db/HumanGutDB-v2.6.fasta
    volumes:
      - /absolute/path/to/your-data:/app/data
      - ./temp_dirs:/app/temp_dirs
    stdin_open: true
    tty: true
```

---

## 🔧 What to Modify

| Field                           | Description                                                                 |
|--------------------------------|-----------------------------------------------------------------------------|
| `/absolute/path/to/your-data`  | Full absolute path to your local directory with `query.fastq` and `blast_db/` |
| `query.fastq`                  | Name of your FASTQ file (must exist inside the mounted `your-data` folder) |
| `blast_db/HumanGutDB-v2.6.fasta` | Path to your BLAST database FASTA file (used to generate the BLAST DB index) |

💡 **Make sure your BLAST database is built before running:**

```bash
makeblastdb -in HumanGutDB-v2.6.fasta -dbtype nucl -out HumanGutDB-v2.6
```

This creates the `.nin`, `.nsq`, and `.nhr` index files required for BLAST.

---

## ▶️ Run the Container

From the same directory as your `docker-compose.yml`, run:

```bash
docker compose up
```

---

## 📁 Output Location

CensuScope writes all output to `/app/temp_dirs` in the container. This is mapped to the `./temp_dirs` folder on your local machine.

---

## 🧼 Stop and Clean Up

```bash
docker compose down
```

To remove volumes too (including contents of `temp_dirs`):

```bash
docker compose down -v
```

---

## 🐛 Troubleshooting (macOS)

If you see a `Mounts denied` error on macOS:

1. Open **Docker Desktop**
2. Go to **Settings > Resources > File Sharing**
3. Add the parent directory of `/your/data` (e.g., `/Users/yourname/GitHub/GW_HIVE`)
4. Click **Apply & Restart**

---

**Comment by penningtonea** [comment]
Will need to move indexed filtered_nt to HIVE server and pull censuscope docker container

**Comment by penningtonea** [comment]
I will move filtered_nt to HIVE 2

**Comment by penningtonea** [comment]
Updated FilteredNT stats - size, date, version, number of sequences

Newest version: 
1.4T Apr 18 01:48 filteredNT_v8.0.fasta 89,505,997

Hadley's version:
1.4T Oct 29 22:52 filteredNT_v8.0.fasta 90,366,870

**Comment by penningtonea** [comment]
FilteredNT_v8.0 is being transferred to HIVE2. Should be ready by tomorrow to decompress and index.

**Comment by cwoodside1278** [comment]
@penningtonea and so the filteredNT file differs from slimNT, correct?

**Comment by cwoodside1278** [comment]
@kee007ney are you stuck on this? Can we discuss this next scrum

**Comment by penningtonea** [comment]
I will start working on the CensuScope docker task next Monday as I am off on Fridays.

**Comment by penningtonea** [comment]
indexing filteredNT_v8.0.1 on HIVE2. Once its done, I will test censuscope with a random sample file moved from Pegasus.

**Comment by penningtonea** [comment]
error:

![Image](https://github.com/user-attachments/assets/99a1b1d8-944d-4184-83e8-1603ba0e8fbe)

**Comment by penningtonea** [comment]
Modified docker-compose.yml for the test data and filteredNT database as shown here: 

![Image](https://github.com/user-attachments/assets/e5fc933e-8bb9-4039-b1e9-d0671343e35d)

**Comment by kee007ney** [comment]
Went over this in depth with Emily yesterday, she's successfully able to run it now.

**Comment by cwoodside1278** [comment]
Emily says that her and Hadley have met on this. Just need to free up more space on HIVE2.

**Comment by cwoodside1278** [comment]
Ask Lam about this so we can add it to HIVE3, made a new ticket
